config/database.yml

```diff
@@ -27,25 +27,25 @@ default: &default
       
 development:
   <<: *default
-  database: PracticalDeveloper_development
+  database: postgres
 
```

Comment:
Hi @iMerica, I was wondering why you changed the default database name, is it just to make the build work with Docker?

Same question goes for the other params. You set a username which, though it's the default user installed with Postgres, might not be the newcomer first choice. Same reasoning goes with the password and the hostname. 

Thanks!

I feel like

Reply:
These are the default credentials on the official [postgres](https://hub.docker.com/_/postgres/) container. I'll replace them with the originals using env vars.

Reply:
why you don't write some kind of entrypoint script, something like this ?
```
cat <<EOF > config/database.yml

default: &default
  adapter: postgresql
  encoding: unicode
  # For details on connection pooling, see rails configuration guide
  # http://guides.rubyonrails.org/configuring.html#database-pooling
  pool: 5
  connect_timeout: 2
  checkout_timeout: 2
  host:     <%= ENV['POSTGRES_HOST'] || 'postgres' %>
  username: <%= ENV['POSTGRES_USER'] || 'postgres' %>
  password: <%= ENV['POSTGRES_PASSWORD'] || 'mysecretpassword' %>
  variables:
      statement_timeout: <%= ENV['STATEMENT_TIMEOUT'] || 2500 %>
      
development:
  <<: *default
  database: <%= ENV['POSTGRES_DATABASE'] || 'PracticalDeveloper_development' %>

test:
  <<: *default
  database: <%= ENV['POSTGRES_DATABASE'] || 'PracticalDeveloper_test' %>

production:
  <<: *default
  database: <%= ENV['POSTGRES_DATABASE'] || 'PracticalDeveloper_production' %>

EOF

set_var() {
VAR_NAME=$1
VAR_VALUE=$2
sed -i "/:${VAR_NAME}/ s~default:.*~default: \"${VAR_VALUE}\"~" Envfile
}

VARS=$(cat Envfile | awk '/variable/{print $2}' | sed 's/[:|,]//g')
for v in $VARS ; do  [[ ${!v} ]] && set_var $v ${!v} ; done

```

Reply:
This is not necessary. The PostgresSQL container supports overrides.

Reply:
And here postgres container ?, so you do not need to interfere with the settings or something else ... look at my MR #342

Reply:
Hi @bdod,

> hardcoding settings it's a bad practice

Please see the [docs](https://hub.docker.com/_/postgres/) for the Postgres container. It is not "hardcoding" when by receiving postgres connection details from the environment.  This is a common 12-factor practice.